state/runtime/evm/instructions_benchmark_test.go

```diff
@@ -0,0 +1,304 @@
+package evm
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	instructionOperation func(c *state)
+)
+
+func BenchmarkStack(b *testing.B) {
+	s, closeFn := getState(&chain.ForksInTime{})
+	defer closeFn()
+
+	op1 := uint256.NewInt(1)
+	op2 := uint256.NewInt(2)
+
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		s.push(*op1)
+		s.push(*op2)
+		s.pop()
+		s.pop()
+	}
+
+	b.StopTimer()
+}
+
+func operationBenchmark(b *testing.B, s *state, op instructionOperation, op1 uint256.Int, op2 uint256.Int) {
+	b.Helper()
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		s.push(op1)
+		s.push(op2)
+		op(s)
+		s.pop()
+	}
+
+	b.StopTimer()
+}
```

Comment:
WDYT (in order not to limit ourselves to only binary operations)?

```suggestion
func operationBenchmark(b *testing.B, s *state, op instructionOperation, operands ...uint256.Int) {
	b.Helper()
	b.ResetTimer()

	for i := 0; i < b.N; i++ {
		for _, op := range operands {
			s.push(op)
		}
		op(s)
		s.pop()
	}

	b.StopTimer()
}
```